kessel-sdk/src/main/java/org/project_kessel/api/rbac/v2/ListWorkspaces.java

```diff
@@ -0,0 +1,124 @@
+            KesselInventoryServiceBlockingStub inventory,
+            SubjectReference subject,
+            String relation,
+            String continuationToken) {
+        return () -> new WorkspaceListIterator(inventory, subject, relation, continuationToken);
+    }
+
+    private static class WorkspaceListIterator implements Iterator<StreamedListObjectsResponse> {
+        private KesselInventoryServiceBlockingStub inventory;
+        private SubjectReference subject;
+        private String relation;
+
+        private String continuationToken;
+        private Iterator<StreamedListObjectsResponse> currentPageIterator;
+        private StreamedListObjectsResponse nextResponse;
+        private boolean hasMoreData;
+
+        public WorkspaceListIterator(
+                KesselInventoryServiceBlockingStub inventory,
+                SubjectReference subject,
+                String relation,
+                String initialToken) {
+            this.inventory = inventory;
+            this.subject = subject;
+            this.relation = relation;
+            this.continuationToken = initialToken;
+            this.currentPageIterator = Collections.emptyIterator();
+            this.hasMoreData = true;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (nextResponse != null) {
+                return true;
+            }
+            return tryToFetchNext();
+        }
+
+        @Override
+        public StreamedListObjectsResponse next() {
+            if (nextResponse == null && !tryToFetchNext()) {
+                throw new NoSuchElementException("No more workspace responses available");
+            }
+            StreamedListObjectsResponse response = nextResponse;
+            nextResponse = null;
+            return response;
+        }
+
+        private boolean tryToFetchNext() {
+            while (true) {
+                if (currentPageIterator.hasNext()) {
+                    nextResponse = currentPageIterator.next();
+                    if (nextResponse.hasPagination()) {
+                        String token = nextResponse.getPagination().getContinuationToken();
+                        if (token != null && !token.isEmpty()) {
+                            this.continuationToken = token;
+                        }
+                    }
+                    // Done feching everything from this page
+                    return true;
+                }
+
+                if (!hasMoreData) {
+                    return false;
+                }
+
+                // Fetch the next page of results.
+                try {
+                    StreamedListObjectsRequest request = buildStreamedListObjectsRequest(subject, relation,
+                            continuationToken);
+                    currentPageIterator = inventory.streamedListObjects(request);
+
+                    if (!currentPageIterator.hasNext()) {
+                        hasMoreData = false;
```

Comment:
Another way to know that we are at the end, is that the `continuationToken` is null - that would save us 1 request and might simplify the code a little bit

Reply:
We kinda have to make two requests because the every response with a workspace has a continuation token, so we need to make 1 extra iteration to see if we're done.

Reply:
IIUC the last object shouldn't have a continuation token - is my understanding incorrect?

Reply:
I thought the same, but even if there is only 1 workspace that one workspace will return with a continuation token.

this is an example response i took from the go sdk
```
Listing workspaces:
2025/10/02 15:18:31 Response: object:{resource_id:"workspace-seed-1" reporter:{type:"rbac"}} pagination:{continuation_token:"ClsKBENMQmkSAi0xEgEwEgASAi0xEgItMRIBMBICLTESAi0xEgEwEgItMRIBMBICLTESATASABICLTESATEaEGRjOGFhZTBhMWU2ZjcxMGIgASoJCgRscnYyEgEx"}
```
there's only 1 workspace

Reply:
Even running against relations-api directly with a lookupresources call gives a continuationtoken for a single response. I'm looking into because I don't feel like it should be